src/panels/lovelace/badges/hui-summary-badge.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  customElement,
+  html,
+  LitElement,
+  TemplateResult,
+  property,
+} from "lit-element";
+import { computeDomain } from "../../../common/entity/compute_domain";
+import "../../../components/ha-label-badge";
+import { HomeAssistant } from "../../../types";
+import { LovelaceBadge } from "../types";
+import { SummaryBadgeConfig } from "./types";
+
+@customElement("hui-summary-badge")
+export class HuiSummaryBadge extends LitElement implements LovelaceBadge {
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property() protected _config?: SummaryBadgeConfig;
+
+  public setConfig(config: SummaryBadgeConfig): void {
+    this._config = {
+      icon: "hass:lightbulb-group",
+      domain: "light",
+      state: "on",
+      ...config,
+    };
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const states: string[] = [];
+
+    for (const entity of Object.keys(this.hass.states)) {
+      if (
+        (!this._config.exclude || !this._config.exclude.includes(entity)) &&
+        computeDomain(entity) === this._config.domain &&
+        this.hass.states[entity].state === this._config.state
+      )
+        states.push(entity);
+    }
```

Comment:
Use filter:

```ts
const states = Object.values(this.hass.states).filter(entity => (!this._config.exclude || !this._config.exclude.includes(entity.entity_id)) &&
        computeDomain(entity.entity_id) === this._config.domain &&
        entity.state === this._config.state));
```

Reply:
We should also add a `shouldUpdate` so we only render and do this if the states changed.